In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("andmed/toorandmed.csv")
pd.set_option('display.max_columns', None)
df.head(1)

course_uuid         code  \
0  eefb3a8d-7669-a5dc-9e13-6f95d719ebdf  LOMR.03.021   

                                       title__en  \
0  Practical Course of Microbiology and Virology   

                              version__title__en  credits  \
0  Practical Course of Microbiology and Virology      6.0   

                           overview__description__en  \
0  Gaining practical skills to work with microbes...   

                  version__overview__description__en  \
0  Gaining practical skills to work with microbes...   

                 overview__learning_outcomes_text_en  \
0  - After the course students are expected to:\n...   

                                   uuid state__code   state__et  state__en  \
0  eefb3a8d-7669-a5dc-9e13-6f95d719ebdf   confirmed  Kinnitatud  confirmed   

           last_update                               title__et  \
0  2024-02-05T11:04:36  Mikrobioloogia ja viroloogia praktikum   

                            general__input_languages  \
0  [{"code": "en", "en": "English", "et": "inglis...   

                     general__structural_unit_shares general__type__code  \
0  [{"coordinating": true, "course_share_percent"...             regular   

  general__type__et general__type__en  \
0     Tavaline aine    Regular course   

                           general__structural_units  \
0  [{"academic": true, "address": "Riia 23, ruumi...   

   additional_info__duration_in_semesters  \
0                                       1   

  additional_info__assessment_scale__code  \
0                                   grade   

  additional_info__assessment_scale__et  \
0        Eristav (A, B, C, D, E, F, mi)   

            additional_info__assessment_scale__en  \
0  differentiated (A, B, C, D, E, F, not present)   

   additional_info__is_vota_course  \
0                             True   

   additional_info__is_continuous_learning_course  \
0                                           False   

                      additional_info__prerequisites  \
0  [{"code": "LOMR.03.010", "credits": 3, "last_u...   

                           overview__description__et  \
0  Praktilise töö kogemuste saamine mikroobidega ...   

                                overview__objectives  \
0  [{"en": "The aim of the Practical Course of Mi...   

                         overview__learning_outcomes  _actions__is_editable  \
0  [{"en": "After the course students are expecte...                  False   

   _actions__is_version_creatable  _actions__is_deletable  \
0                           False                   False   

                        overview__objectives_text_en  \
0  - The aim of the Practical Course of Microbiol...   

                          version__uuid version__state__code  \
0  b8342470-93bc-fd9b-65a6-42b9ea4d937a            confirmed   

  version__state__et version__state__en version__last_update  \
0         Kinnitatud          confirmed  2026-01-15T19:19:53   

                       version__title__et  version__credits  \
0  Mikrobioloogia ja viroloogia praktikum               2.0   

  version__credit_type  version__target__year__code version__target__year__et  \
0                 ECTS                       2025.0                 2025/2026   

  version__target__year__en version__target__semester__code  \
0                 2025/2026                          autumn   

  version__target__semester__et version__target__semester__en  \
0                         sügis                        autumn   

  version__target__study_type__code version__target__study_type__et  \
0                          fulltime                        päevaõpe   

  version__target__study_type__en version__target__language__code  \
0                 regular studies                              et   

  version__target__language__et version__target__language__en  \
0                    eesti keel                      Estonian   

   version__target__part__code version__target__part__et  \
0                      

In [3]:
# len(df.code)
# len(df.columns)
df.shape

(3031, 223)

### Arvutada tunnuste ja andmepunktide arv toorandmestikus:

- 223 tunnust ja 3031 andmepunkte.


### Vaadata põhjalikult läbi üks andmerida

In [4]:
df['null_count (lisa)'] = df.isnull().sum(axis=1)
df.sort_values("null_count (lisa)", ascending=False).iloc[0]["code"]


'HVLC.02.016'

In [5]:
rida = df[df.code=="HVLC.02.016"]
puhas = rida.iloc[0]
puhas = puhas.dropna()

In [6]:
puhas = puhas.to_frame(name="value")
puhas

value
course_uuid                                                  a637520e-6c71-aacf-628d-96ceffa6dcef
code                                                                                  HVLC.02.016
title__en                                                   Trainee Practice in Teaching Antiques
credits                                                                                      11.0
overview__description__en                                                                       .
overview__learning_outcomes_text_en                                                           - .
uuid                                                         a637520e-6c71-aacf-628d-96ceffa6dcef
state__code                                                                             confirmed
state__et                                                                              Kinnitatud
state__en                                                                               confirmed
last_update                                                                   2025-01-17T08:50:46
title__et                                                       Antiikkeelte õpetaja põhipraktika
general__input_languages                        [{"code": "et", "en": "Estonian", "et": "eesti...
general__structural_unit_shares                 [{"coordinating": true, "course_share_percent"...
general__type__code                                                                       regular
general__type__et                                                                   Tavaline aine
general__type__en                                                                  Regular course
general__structural_units                       [{"academic": true, "code": "HVLC02", "level":...
additional_info__duration_in_semesters                                                          1
additional_info__assessment_scale__code                                                      pass
additional_info__assessment_scale__et                                 Eristamata (arv, m.arv, mi)
additional_info__assessment_scale__en                non-differentiated (pass, fail, not present)
additional_info__is_vota_course                                                              True
additional_info__is_continuous_learning_course                                              False
overview__description__et                                                                       .
overview__objectives                                                     [{"en": ".", "et": "."}]
overview__learning_outcomes                                              [{"en": ".", "et": "."}]
_actions__is_editable                                                                       False
_actions__is_version_creatable                                                              False
_actions__is_deletable                                                                      False
overview__objectives_text_en                                                                  - .
null_count (lisa)                                                                             192

In [7]:
puhas.iat[17, 0]


'[{"academic": true, "code": "HVLC02", "level": 3, "name": {"en": "Department of Classical Studies", "et": "klassikalise filoloogia osakond"}, "parent_code": "HVLC", "supports_continuous_learning": false}]'

### Üks andmerida + lühike paragrahv üldistest tähelepanekutest andmerea kohta
192 veergu 223 st (86%) on tühjad, (võib eeldada, et need veerud on kohustuslikud)
Päriselt täidetud info piirdub tehnilise miinimumiga: 2 uuid + 1 code, 5 tõeväärtust, info hindamise kohta ning tõlged nimetuse ja oleku jaoks.
Väljad kirjelduse jaoks pole tühjad, samas pole nad informatiivsed (nt: “.”, “- .”, “.”, “- .”)